pkg/apis/autopilot/v1alpha1/autopilotrule.go

```diff
@@ -100,6 +104,72 @@ type AutopilotRuleSpec struct {
 	ActionsCoolDownPeriod int64 `json:"actionsCoolDownPeriod,omitempty"`
 }
 
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AutopilotRuleObject represents a particular object that is being monitored by autopilot. This is primarily used
+// for status purposes
+type AutopilotRuleObject struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	// Status is the status of an object monitored by an autopilot rule
+	Status AutopilotRuleObjectStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AutopilotRuleObjectList is a list of AutopilotRule objects in Kubernetes
+type AutopilotRuleObjectList struct {
+	meta.TypeMeta `json:",inline"`
+	meta.ListMeta `json:"metadata,omitempty"`
+
+	Items []AutopilotRuleObject `json:"items"`
```

Comment:
@adityadani heads up. I had to convert status is a list of items instead of a single status. This is so that we can have a list of recent statuses for an object instead of just the most recent one.

Reply:
how does the list keep an order?

Reply:
sounds good to me. 
@trierra  Sort it by status.LastProcessTimestamp and then regularly prune older ones from the slice ?

Reply:
Yeah the order is maintained at the autopilot side

Reply:
This how it will show up as an example

```
kubectl describe aro
Name:         0d46fd08-6ce6-4b97-922b-5e438d795a29
Namespace:
Labels:       rule=pool-expand
Annotations:  <none>
API Version:  autopilot.libopenstorage.org/v1alpha1
Kind:         AutopilotRuleObject
Metadata:
  Creation Timestamp:  2020-01-15T18:28:43Z
  Generation:          5
  Resource Version:    9085879
  Self Link:           /apis/autopilot.libopenstorage.org/v1alpha1/autopilotruleobjects/0d46fd08-6ce6-4b97-922b-5e438d795a29
  UID:                 dc60d570-37c4-11ea-bde4-005056a3f8c1
Status:
  Items:
    Last Process Timestamp:  2020-01-15T18:28:43Z
    Message:                 rule: pool-expand:0d46fd08-6ce6-4b97-922b-5e438d795a29 transition from Normal => Triggered
    State:                   Triggered
    Last Process Timestamp:  2020-01-15T18:29:02Z
    Message:                 rule: pool-expand:0d46fd08-6ce6-4b97-922b-5e438d795a29 transition from Triggered => ActiveActionsPending
    State:                   ActiveActionsPending
    Last Process Timestamp:  2020-01-15T18:29:04Z
    Message:                 rule: pool-expand:0d46fd08-6ce6-4b97-922b-5e438d795a29 transition from ActiveActionsPending => ActiveActionsInProgress
    State:                   ActiveActionsInProgress
    Last Process Timestamp:  2020-01-15T18:29:19Z
    Message:                 rule: pool-expand:0d46fd08-6ce6-4b97-922b-5e438d795a29 transition from ActiveActionsInProgress => ActiveActionsTaken
    State:                   ActiveActionsTaken
    Last Process Timestamp:  2020-01-15T18:34:20Z
    Message:                 rule: pool-expand:0d46fd08-6ce6-4b97-922b-5e438d795a29 transition from ActiveActionsTaken => Normal
    State:                   Normal
Events:                      <none>
```